src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -66,32 +70,101 @@ export default () => {
 
+                deferredUpdateBeforeReconnect = updateParams;
+
                 // Since this is a full reconnectApp, we'll not apply the updates we received - those will come in the reconnect app request.
                 canUnpauseQueuePromise = App.finalReconnectAppAfterActivatingReliableUpdates();
             } else {
                 // The flow below is setting the promise to a getMissingOnyxUpdates to address flow (2) explained above.
+
+                // Get the number of deferred updates before adding the new one
+                const existingDeferredUpdatesCount = deferredUpdates.length;
+                // Add the new update to the deferred updates
+                deferredUpdates.push(updateParams);
+
+                // If there are already deferred updates, then we don't need to fetch the missing updates again
+                if (existingDeferredUpdatesCount > 0) {
+                    return;
+                }
+
                 console.debug(`[OnyxUpdateManager] Client is behind the server by ${Number(previousUpdateIDFromServer) - lastUpdateIDAppliedToClient} so fetching incremental updates`);
                 Log.info('Gap detected in update IDs from server so fetching incremental updates', true, {
                     lastUpdateIDFromServer,
                     previousUpdateIDFromServer,
                     lastUpdateIDAppliedToClient,
                 });
+
+                // Get the missing Onyx updates from the server
                 canUnpauseQueuePromise = App.getMissingOnyxUpdates(lastUpdateIDAppliedToClient, previousUpdateIDFromServer);
             }
 
-            canUnpauseQueuePromise.finally(() => {
-                OnyxUpdates.apply(updateParams).finally(() => {
+            // This function will apply the deferred updates in order after the missing updates are fetched
+            function applyDeferredUpdates() {
+                // If lastUpdateIDAppliedToClient is null, then no updates were applied,
+                // therefore something must have gone wrong and we should handle the problem.
+                if (!lastUpdateIDAppliedToClient) {
+                    // TODO: Handle this case
+                    return;
+                }
+
+                function unpauseQueueAndReset() {
                     console.debug('[OnyxUpdateManager] Done applying all updates');
                     Onyx.set(ONYXKEYS.ONYX_UPDATES_FROM_SERVER, null);
                     SequentialQueue.unpause();
+                }
+
+                // If "updateAfterReconnectApp" is set, it means that case (1) was triggered and we only need to apply the one update, not the deferred ones.
+                if (deferredUpdateBeforeReconnect) {
+                    OnyxUpdates.apply(deferredUpdateBeforeReconnect).finally(unpauseQueueAndReset);
+                    deferredUpdateBeforeReconnect = undefined;
+                    return;
```

Comment:
You should not return here, since `deferredUpdates` could still exist and would not be applied in this case

Reply:
I think i might not understand case (1) correctly. Why would we not receive the `lastUpdateID` from Onyx updates in the first place? Isn't this something thats always provided with updates?

Also, what does the `ReconnectApp` query achieve in case (1)?

Reply:
can you point me to what you're talking about when you say `Case (1)`?

Reply:
oh sorry, i meant what's described here: https://github.com/Expensify/App/blob/9a1f43c286c599b0a694fad3ad5bd9ef7c0b309a/src/libs/actions/OnyxUpdateManager.ts#L57-L67

Reply:
Ohh I see.

**Reliable** OnyxUpdate is something that is not enabled for everyone. So some users will log in and never receive previousUpdateID and lastUpdateID.  When we do enable it for them, once they receive the first request with lastUpdateID, we need to do a full reconnect to confirm that their client is in the right state before we start applying the updates.

Reply:
What I meant with `You should not return here, since deferredUpdates could still exist and would not be applied in this case` is that while you're executing the full reconnect, it's possible that you're still gonna receive updates while that is running (since that's a slow GET). So we should still queue them up as we do for the getMissingOnyxMessages

Reply:
Got it thanks, for the explanation. I already applied the change, so i'm not early-returning in this case..